src/ui/Analytics/Analytics.ts

```diff
@@ -371,6 +373,42 @@ export class Analytics extends Component {
     this.client.setOriginContext(originContext);
   }
 
+  /**
+   * Re-enables analytics if they were previously disabled.
+   */
+  public enable() {
+    if (!this.disabled) {
+      console.warn('Analytics are already enabled.');
+      return;
+    }
+    super.enable();
+    this.initializeAnalyticsClient();
+    this.resolveQueryController().historyStore = new HistoryStore();
+  }
+
+  /**
+   * Removes the analytics tracking cookie and clears the actions history.
+   */
+  public clearLocalData() {
+    this.client.endpoint.forgetVisitorId();
+    this.resolveQueryController().historyStore.clear();
+  }
+
+  /**
+   * Disables analytics and clears local data by running [`clearLocalData`]{@link Analytics.clearLocalData}.
+   */
+  public disable() {
+    if (this.disabled) {
+      console.warn('Analytics are already disabled.');
+      return;
+    }
+    super.disable();
+    this.clearLocalData();
+    this.client.endpoint.kill();
```

Comment:
Related to my comment about, I don't think there is a need to disable/kill the endpoint if we are replacing the client to a Noop client on the next line.